src/index.js

```diff
@@ -116,25 +86,93 @@ const showResult = function () {
   timerElement.classList.toggle("hide");
 };
 
+const saveResult = async () => {
+  const user = auth.currentUser;
+  if (user) {
+    try {
+      await addDoc(collection(db, "results"), {
+        userId: user.email,
+        result: totalScore,
+        timestamp: new Date(),
+      });
+      console.log("Wynik został zapisany do Firestore.");
+    } catch (error) {
+      console.error("Wystąpił błąd podczas zapisywania wyniku:", error);
+    }
+  }
+};
+
 const onSubmit = () => {
   addPoints();
   showResult();
+  saveResult();
 };
 
-startBtn.addEventListener("click", startQuiz);
+let resultsVisible = false;
+const showPreviousResult = async () => {
+  const user = auth.currentUser;
+  if (user) {
+    try {
+      const querySnapshot = await getDocs(
+        query(collection(db, "results"), where("userId", "==", user.email))
+      );
+      const results = [];
+      querySnapshot.forEach((doc) => {
+        results.push(doc.data());
+      });
+      if (results.length > 0) {
+        displayResults(results);
+      } else {
+        displayNoResultsMessage();
+      }
+    } catch (error) {
+      console.error("Wystąpił błąd podczas pobierania wyników:", error);
+    }
+  }
+};
 
-submitBtn.addEventListener("click", onSubmit);
+const displayResults = (results) => {
+  const previousResultSection = document.getElementById(
+    "previous-result-section"
+  );
+  previousResultSection.innerHTML = "";
+
+  results.forEach((result) => {
+    const resultElement = document.createElement("div");
+    resultElement.innerHTML = `Result: ${
+      result.result
+    } / 21<br>Date: ${result.timestamp.toDate()}`;
```

Comment:
21 is a magic number here. Giving it a name like "maxScore" could improve the readability.